kv/option.go

```diff
@@ -96,6 +96,14 @@ const (
 	// ScanBatchSize set the iter scan batch size.
 	ScanBatchSize
 	// TxnSource set the source of this transaction.
+	// For now, it can represent the transaction is created by a TiCDC or Lossy DDL reorg Backfill job.
+	// 1. Reserve 1 - 15 for TiCDC to implement BDR synchronization.
+	// 2. Reserve 1024 - `To Be Determined` for Lossy DDL Backfill job.
+	//    For now, we only use 1024 for column reorg backfill job.
+	// The reason for having these spans is to keep some room for future expansion of it.
+	// We only can set one source for a transaction, so please make sure the different features which
+	// use this option don't conflict with each other.
+	// Or find a way to set multiple sources for a transaction in an uint64.
```

Comment:
Do you think we need to redesign it?
If we want to add more sources in the future, some sources may conflict with other sources.
For example, if we add a new feature, it depends on a new 2048-2060 range. At the same time, we have a flag for TiKV to indicate what TiKV should do when it gets this kind of txn.
If users also enable the BDR feature, then it conflicts. Because we can only set one `txn_option` for one txn.

So I suggest we use bits to represent different sources.
For example:
```go
package main

import "fmt"

const (
	stateABits  = 8 // Number of bits used for state A
	stateBBits  = 4 // Number of bits used for state B
	stateCBits  = 4 // Number of bits used for state C
	stateAMax   = (1 << stateABits) - 1
	stateBMax   = (1 << stateBBits) - 1
	stateCMax   = (1 << stateCBits) - 1
	stateBShift = stateABits
	stateCShift = stateABits + stateBBits
)

func main() {
	var flags uint64 = 0 // Initial value is 0

	// Set state A to 7
	setStateA(&flags, 7)

	// Set state B to 11
	setStateB(&flags, 11)

	// Set state C to 3
	setStateC(&flags, 3)

	// Check if state A, B, and C are set
	if isStateASet(flags) {
		fmt.Println("State A is set")
	}
	if isStateBSet(flags) {
		fmt.Println("State B is set")
	}
	if isStateCSet(flags) {
		fmt.Println("State C is set")
	}

	// Get the values of state A, B, and C
	a := getStateA(flags)
	b := getStateB(flags)
	c := getStateC(flags)
	fmt.Printf("State A: %d, State B: %d, State C: %d\n", a, b, c)
}

// Set the value of state A
func setStateA(flags *uint64, value uint64) {
	if value > stateAMax {
		value = stateAMax
	}
	*flags |= value
}

// Set the value of state B
func setStateB(flags *uint64, value uint64) {
	if value > stateBMax {
		value = stateBMax
	}
	*flags |= value << stateBShift
}

// Set the value of state C
func setStateC(flags *uint64, value uint64) {
	if value > stateCMax {
		value = stateCMax
	}
	*flags |= value << stateCShift
}

// Check if state A is set
func isStateASet(flags uint64) bool {
	return (flags & stateAMax) != 0
}

// Check if state B is set
func isStateBSet(flags uint64) bool {
	return (flags & (stateBMax << stateBShift)) != 0
}

// Check if state C is set
func isStateCSet(flags uint64) bool {
	return (flags & (stateCMax << stateCShift)) != 0
}

// Get the value of state A
func getStateA(flags uint64) uint64 {
	return flags & stateAMax
}

// Get the value of state B
func getStateB(flags uint64) uint64 {
	return (flags >> stateBShift) & stateBMax
}

// Get the value of state C
func getStateC(flags uint64) uint64 {
	return (flags >> stateCShift) & stateCMax
}
```

Then we can use A for BDR, B for lossy DDL.
Check compatibility:
1. Old TiDB will set txn_option to 1-15.
2. Old TiKV will drop the event if txn_option is not 0.
3. TiUP will update tikv first, so we need to make sure that new TiKV can handle the txn_option set by old TiDB.
4. The new TiKV will get the first 8 bits for the BDR function, because the old TiDB only allows to set 1-15 to txn_option. So it is compatible.

Reply:
LGTM